packages/target-electron/static/webxdc-preload.js

```diff
@@ -291,17 +298,29 @@ class RealtimeListener {
           ipcRenderer.invoke('webxdc.exit')
         } catch (error) {
           loadingDiv.remove()
-          iframe.src = 'index.html'
+          iframe.src = locationUrl !== '' ? locationUrl : 'index.html'
           iframe.contentWindow.window.addEventListener(
             'keydown',
             keydown_handler
           )
+          connectionsFilled = true
         }
       } catch (error) {
         console.log('error loading, should crash/close window', error)
         ipcRenderer.invoke('webxdc.exit')
       }
     },
+    /**
+     * called via webContents.executeJavaScript
+     */
+    setLocationUrl(base64EncodedHref) {
+      locationUrl = Buffer.from(base64EncodedHref, 'base64').toString('utf8')
+      if (locationUrl && locationUrl !== '' && connectionsFilled) {
+        // if connectionsFilled is false, the url is loaded after
+        // the connections are filled
+        window.frames[0].window.location = locationUrl
```

Comment:
```suggestion
        document.getElementById('frame').contentWindow.location = locationUrl
```

This is more concrete. Feel free to ignore

Reply:
Maybe in a later change, I would like to refactor some more stuff later :-)